test/testscript.go

```diff
@@ -0,0 +1,155 @@
+package test
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"text/template"
+
+	"github.com/rogpeppe/go-internal/testscript"
+)
+
+type ProjectTest struct {
+	t          *testing.T
+	params     testscript.Params
+	setupFuncs []func(*testscript.Env) error
+	projectDir string
+}
```

Comment:
I called this struct a "ProjectTest" as the idea is it helps the user test a given project that have files/templates/artifacts/make targets we care about. 

Feel free to suggest naming updates.